cryostatio/cryostat-operator
# chore(api): remove ClusterCryostat CRD

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #732 

## Description of the change:
* This removes all versions of the ClusterCryostat CRD and any controller code was associated with it.
* The tests for conflicting Cryostat and ClusterCryostat CRs have been reworked to test scenarios where two Cryostat CRs may be using the same namespace. In all cases, there should be no conflicts anymore.
* Added tests for the RoleBinding finalizer logic that was previously missed.
* Merged the multi-namespace.md doc into config.md

## Motivation for the change:
* The Cryostat CRD is now capable of handling multi-namespace Cryostat instances. The ClusterCryostat CRD no longer serves a purpose and adds unnecessary maintenance and testing burden.

## How to manually test:
1. Build/deploy a bundle, you should no longer see any ClusterCryostat options and everything else should work as before.
